lib/src/command/upgrade.dart

```diff
@@ -44,14 +61,163 @@ class UpgradeCommand extends PubCommand {
       log.warning(log.yellow(
           'The --packages-dir flag is no longer used and does nothing.'));
     }
-    await entrypoint.acquireDependencies(SolveType.UPGRADE,
-        useLatest: argResults.rest,
-        dryRun: argResults['dry-run'],
-        precompile: argResults['precompile']);
+
+    if (argResults.wasParsed('breaking')) {
+      final useLatest = argResults.rest;
+
+      final rootPubspec = entrypoint.root.pubspec;
+      final resolvablePubspec =
+          _stripVersionConstraints(rootPubspec, useLatest: useLatest);
+
+      Map<String, PackageId> resolvablePackages;
+      await log.spinner('Resolving', () async {
+        resolvablePackages = await _tryResolve(resolvablePubspec);
+      }, condition: _shouldShowSpinner);
+
+      final dependencyChanges = <PackageRange, PackageId>{};
+      final allDependencies = [
+        ...rootPubspec.dependencies.values,
+        ...rootPubspec.devDependencies.values
+      ];
+
+      for (var package in allDependencies) {
+        final resolvedPackage = resolvablePackages[package.name];
+
+        if (resolvedPackage != null &&
+            !package.constraint.allows(resolvedPackage.version) &&
+            package.source is HostedSource) {
+          dependencyChanges[package] = resolvedPackage;
+        }
+      }
+
+      if (!argResults['dry-run']) {
+        await _writeToPubspec(dependencyChanges);
+        await Entrypoint.current(cache).acquireDependencies(SolveType.UPGRADE);
```

Comment:
Here you don't pass --dry-run and --precompile.

Is it possible to do the acquireDependencies after `if (argResults.wasParsed('breaking')) {...}` So it is the same no matter what?

Reply:
I'll modify the outer acquireDependencies to:

```dart
await Entrypoint.current(cache).acquireDependencies(SolveType.UPGRADE,
        useLatest: argResults.rest,
        dryRun: argResults['dry-run'],
        precompile: argResults['precompile']);
```

which works for both and does not seem like it will result in performance penalties, but I'm not too sure about that.